packages/server/routes/api/auth.js

```diff
@@ -107,4 +114,88 @@ router.get("/google/callback", passport.authenticate("google", {
   response.redirect(`icebreak://login?token=${newToken}`);
 });
 
-module.exports = router;
\ No newline at end of file
+router.post("/register", async (request, response) => {
+
+  try{
+    const { email, password, } = request.body;
+    
+    let emailCheckRegex = /^[\w-\.]+@([\w-]+\.)+[\w-]{2,4}$/g;
+
+    if(!emailCheckRegex.test(email)){ // check if email is valid, doesn't include or no spaces
+      throw new Error("Email is invalid.");
+    }
+
+    const ifExist = await user.getUserByEmail(email);
+
+    if(ifExist?.email === email){ // check if email is already in the database
+      throw new Error("User already exists with this email.");
+    }
+
+    const saltRounds = 10;
+
+    bcrypt.genSalt(saltRounds, function(err, salt) {
+      bcrypt.hash(password, salt, async function(err, hash) { // encrypt the user password using bcrypt
+
+        // create unique User ID as bytes (18 byte)
+        const user_id = uniqid();
+
+        await postgres.query(`
+          INSERT INTO users (user_id, first_name, last_name, email, avatar, password)
+          VALUES ('${user_id}', 'firstName', 'lastName', '${email}', 'avatar', '${password}');
+        `); // create new user in DB
+
+        const newToken = token.generate({ email, hash}); // create a signed jwt token
+
+        response.send({  // send jwt token
+          success: true,
+          newToken
+          }
+        ); 
+      });
+    });
+  }
+  catch(error){
+    response.status(403).send({
+      message: error.message,
+      success: false
+    });
+  }
+});
+
+
+router.post("/login", async (request, response) => {
+  
+  try{
+    
+    // Get user input
+    const { email, password } =  request.body;
+    const ifExist = await user.getUserByEmail(email);
+    const hashedPassword = ifExist.password;
+
+    // Validate credentials
+    if(!(email && password)){
+      response.status(403).send("All inputs are required");
```

Comment:
When the consuming the API expected response in JSON, it should be all in JSON format including the data or it could cause unexpected errors on the client-side.